builtins/src/lib.rs

```diff
@@ -0,0 +1,433 @@
+        entrypoint: solana_bpf_loader_program::Entrypoint::vm,
+        native_cu_cost: solana_bpf_loader_program::DEFAULT_LOADER_COMPUTE_UNITS,
+    }),
+    testable_prototype!(BuiltinPrototype {
+        core_bpf_migration_config: None,
+        name: solana_bpf_loader_upgradeable_program,
+        enable_feature_id: None,
+        program_id: bpf_loader_upgradeable::id(),
+        entrypoint: solana_bpf_loader_program::Entrypoint::vm,
+        native_cu_cost: solana_bpf_loader_program::UPGRADEABLE_LOADER_COMPUTE_UNITS,
+    }),
+    testable_prototype!(BuiltinPrototype {
+        core_bpf_migration_config: None,
+        name: compute_budget_program,
+        enable_feature_id: None,
+        program_id: solana_sdk_ids::compute_budget::id(),
+        entrypoint: solana_compute_budget_program::Entrypoint::vm,
+        native_cu_cost: solana_compute_budget_program::DEFAULT_COMPUTE_UNITS,
+    }),
+    BuiltinPrototype {
+        core_bpf_migration_config: Some(CoreBpfMigrationConfig {
+            source_buffer_address: buffer_accounts::address_lookup_table_program::id(),
+            upgrade_authority_address: None,
+            feature_id: solana_feature_set::migrate_address_lookup_table_program_to_core_bpf::id(),
+            migration_target: CoreBpfMigrationTargetType::Builtin,
+            datapoint_name: "migrate_builtin_to_core_bpf_address_lookup_table_program",
+        }),
+        name: "address_lookup_table_program",
+        enable_feature_id: None,
+        program_id: solana_sdk_ids::address_lookup_table::id(),
+        entrypoint: solana_address_lookup_table_program::processor::Entrypoint::vm,
+        native_cu_cost: solana_address_lookup_table_program::processor::DEFAULT_COMPUTE_UNITS,
+    },
+    testable_prototype!(BuiltinPrototype {
+        core_bpf_migration_config: None,
+        name: zk_token_proof_program,
+        enable_feature_id: Some(feature_set::zk_token_sdk_enabled::id()),
+        program_id: solana_sdk_ids::zk_token_proof_program::id(),
+        entrypoint: solana_zk_token_proof_program::Entrypoint::vm,
+        native_cu_cost: 0, // TODO, these are per-instruction.
+    }),
+    testable_prototype!(BuiltinPrototype {
+        core_bpf_migration_config: None,
+        name: loader_v4,
+        enable_feature_id: Some(feature_set::enable_program_runtime_v2_and_loader_v4::id()),
+        program_id: solana_sdk_ids::loader_v4::id(),
+        entrypoint: solana_loader_v4_program::Entrypoint::vm,
+        native_cu_cost: solana_loader_v4_program::DEFAULT_COMPUTE_UNITS,
+    }),
+    testable_prototype!(BuiltinPrototype {
+        core_bpf_migration_config: None,
+        name: zk_elgamal_proof_program,
+        enable_feature_id: Some(feature_set::zk_elgamal_proof_program_enabled::id()),
+        program_id: solana_sdk_ids::zk_elgamal_proof_program::id(),
+        entrypoint: solana_zk_elgamal_proof_program::Entrypoint::vm,
+        native_cu_cost: 0, // TODO, these are per-instruction.
```

Comment:
@tao-stones the two ZK builtins are going to be tricky since they are metered per-instruction.

Reply:
Yea, I don't know what to do with ZK builtins, they are per instruction and they are consume much more CUs than other regular builtins. We could consider them as "not builtins" so to allocate 200K by default (Assume users of ZK will set cu-limit sensibly); or we can speed up SIMD-198 to make all remaining builtins per instruction as well (tho that does not address the issue that original builtins consumes hundreds of CUs, while ZK takes thousands of CUs)

Reply:
Well, you originally didn't have them in the list, so should they not be included here at all? Or is setting them to zero okay?

Reply:
There was no ZK when the list was created :) I think ZK builtins should be included in the list with "MAX(per instruction cost)"; then:
- before SIMD-170 (eg, current behavior): cost mode reserve enough CUs for all its instructions, maybe a bit over reserving for some, but that is not new. VM's CU Meter always give default 200K for it.
- after SIMD-170, both cost model and CU meter don't care about `native_cu_cost` anymore, they will give 3K CU for it, if user didn't request CU Limit, therefore fails TX.

Reply:
wdyt @jstarry

Reply:
I don't really follow what you're suggesting we do in your comment above. But in any case, this PR should definitely not be changing the static list of builtins without a feature gate.

Reply:
Not disagree that the static list shouldn't be changed without feature gate. But technically can still do that before #3799 feature gate is activated.

Reply:
Yes we could but I don't a crate refactoring PR is the right time do that. If we want to update the list before SIMD-0170 is activated we would need to backport the change to v2.0 and v2.1 and update the list of builtin id's included in the SIMD-0170 proposal details.